src/Widgets/MultitaskingView.vala

```diff
@@ -321,7 +321,10 @@ namespace Gala {
             GestureTracker.OnEnd on_animation_end = (percentage, cancel_action, calculated_duration) => {
                 workspace_gesture_tracker.enabled = false;
 
-                var duration = is_nudge_animation ? (AnimationDuration.NUDGE / 2) : calculated_duration;
+                var duration = is_nudge_animation ?
+                               (uint)(AnimationDuration.NUDGE / 2) :
+                               (uint)(calculated_duration.clamp (0, AnimationDuration.WORKSPACE_SWITCH));
```

Comment:
Couldn't this change the animation duration if the calculated values is greater than `AnimationDuration.WORKSPACE_SWITCH`?

Reply:
This is done to ensure that the `duration` is within the range of `AnimationDuration`.  Otherwise it could (in theory) be arbitrarily long.  But I can remove this change as it is not strictly in scope.

Reply:
It was also to fix the warning `Unsafe conversion from `int' to `Gala.AnimationDuration'` but I can try to do that without limiting calculated duration.

Reply:
> This is done to ensure that the duration is within the range of AnimationDuration. Otherwise it could (in theory) be arbitrarily long

That's intended, the gesture handler takes care of the animation duration and should guarantee a reasonable animation time.

> It was also to fix the warning Unsafe conversion from 'int' to 'Gala.AnimationDuration'

If this warning is present in all uses of the signal, we could emit an unsigned int instead of an int to fix all warnings.